tests/sharingTest.js

```diff
@@ -408,83 +736,71 @@ describe('Main: Currently testing file/folder sharing,', function () {
+      // https://github.com/pact-foundation/pact-js/issues/604
       it('checking method : getShare with existent share', async function () {
-        const provider = createProvider()
-        await getCapabilitiesInteraction(provider)
-        await getCurrentUserInformationInteraction(provider)
-        for (let i = 0; i < config.testFiles.length; i++) {
-          await getShareInteraction(provider, config.testFilesId[i])
-        }
+        for (let i = 0; i < testFiles.length; i++) {
+          const provider = createProvider()
+          await getCapabilitiesInteraction(provider, sharer, sharerPassword)
+          await getCurrentUserInformationInteraction(provider, sharer, sharerPassword)
+          await getShareInteraction(provider, '(group share)', 1, testFiles[i])
 
-        return provider.executeTest(async () => {
-          const oc = createOwncloud()
-          await oc.login()
-          for (const file in sharedFiles) {
-            await oc.shares.getShare(sharedFiles[file]).then(share => {
+          return provider.executeTest(async () => {
+            const oc = createOwncloud(sharer, sharerPassword)
+            await oc.login()
+            await oc.shares.getShare(testFilesId[i]).then(share => {
               expect(typeof (share)).toBe('object')
               expect(share.getId()).toBeGreaterThan(-1)
             }).catch(error => {
               expect(error).toBe(null)
             })
-          }
-        })
+          })
+        }
       })
+    })
 
-      it('checking method : getShares for shared file', async function () {
-        const allIDs = []
-        for (var file in sharedFiles) {
-          allIDs.push(sharedFiles[file])
-        }
+    it('checking method : getShares for shared file', async function () {
+      const provider = createProvider()
+      await getCapabilitiesInteraction(provider, sharer, sharerPassword)
+      await getCurrentUserInformationInteraction(provider, sharer, sharerPassword)
 
-        const provider = createProvider()
-        await getCapabilitiesInteraction(provider)
-        await getCurrentUserInformationInteraction(provider)
-        for (let i = 0; i < testFiles.length; i++) {
-          await getSharesInteraction(provider, `get shares of '${testFiles[i]}' (group share)`, 1, testFiles[i])
-        }
+      for (const file of testFiles) {
+        await getSharesInteraction(provider, file)
+      }
 
-        return provider.executeTest(async () => {
-          const oc = createOwncloud()
-          await oc.login()
-          for (file in sharedFiles) {
-            await oc.shares.getShares(file).then(shares => {
-              expect(shares.constructor).toBe(Array)
-              let flag = 0
-              for (let i = 0; i < shares.length; i++) {
-                const share = shares[i]
-                if (allIDs.indexOf(share.getId()) > -1) {
-                  flag = 1
-                }
-              }
```

Comment:
why did you remove this part of the test

Reply:
I directly counted share count and asserted it, because at given state three shares are created